### PR TITLE
Sync by subscription manager uuid

### PR DESCRIPTION
**Pull request by ShimShtein**

### CODE REVIEW

**Review by laviro — COMMENTED** [review]
thanks @ShimShtein,

I tested insights recommendation sync and woks as expected,
though while testing inventory sync, I got the following error:
```bash
ActiveRecord::RecordNotUnique: PG::UniqueViolation: ERROR:  duplicate key value violates unique constraint "index_insights_facets_on_host_id"
DETAIL:  Key (host_id)=(1) already exists.
```
see more: http://pastebin.test.redhat.com/952029

code wise looks good, @adamruzicka can you also take a look? thanks!

**Review by adamruzicka — COMMENTED** [review]
Not a huge fan of ActiveSupport::Callbacks, but as long as it works.

File: lib/insights_cloud/async/insights_full_sync.rb [hunk 1]
```diff
@@ -6,14 +6,22 @@ class InsightsFullSync < ::ApplicationJob
       include ::ForemanRhCloud::CloudAuth
 
       def perform
+        # This can be turned off when we enable automatic status syncs
+        # This step will query cloud inventory to retrieve inventory uuids for each host
+        ForemanTasks.sync_task(InventorySync::Async::InventoryHostsSync)
```
[the remark this answers is not in the document]
Comment (ShimShtein) [reply lib/insights_cloud/async/insights_full_sync.rb:11]: I will refactor this job into task soon enough, unless you think it has to be done in this PR

**Comment by ShimShtein** [comment]
Resolved the duplicate key issue. I was missing a splat operator:
instead of 
```
uuids_hash.except(existing_facets)
```
it should be
```
uuids_hash.except(*existing_facets)
```

File: lib/insights_cloud/async/insights_full_sync.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (adamruzicka) [reply lib/insights_cloud/async/insights_full_sync.rb:11]: If that's on the roadmap then it is fine by me

**Review by laviro — COMMENTED** [review]
Thanks @ShimShtein, that issue was resolved!

but still while testing the inventory sync with this PR, I got no synced hosts with a subscription where there should be one host synced.

with this PR:
Screenshot (84)

without:
Screenshot (83)

**Review by laviro — APPROVED** [review]
Sorry, the last attempt was with old centos that had a subscription :D
added an RHEL host and now it works as usual, thanks @ShimShtein!

**Pull request merged**